#! /bin/sh /usr/share/dpatch/dpatch-run
## 01-opendir.patch.dpatch by  <jbouse@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix for opendir() bug #484876 provided by Harald Bongartz

@DPATCH@
diff -urNad apt-mirror~/apt-mirror apt-mirror/apt-mirror
--- apt-mirror~/apt-mirror	2007-11-30 17:21:29.000000000 -0500
+++ apt-mirror/apt-mirror	2009-12-23 12:28:56.000000000 -0500
@@ -518,14 +518,14 @@
     my $dir = shift;
     my $is_needed = 0;
     return 1 if $skipclean{$dir};
-    opendir(DIR, $dir) or die "apt-mirror: can't opendir $dir: $!";
-    foreach (grep { !/^\.$/ && !/^\.\.$/ } readdir(DIR)) {
+    opendir(my $dir_h, $dir) or die "apt-mirror: can't opendir $dir: $!";
+    foreach (grep { !/^\.$/ && !/^\.\.$/ } readdir($dir_h)) {
 	my $item = $dir . "/". $_;
 	$is_needed |= process_directory($item)	if -d $item && ! -l $item;
 	$is_needed |= process_file($item)	if -f $item;
 	$is_needed |= process_symlink($item)	if -l $item;
     }
-    closedir DIR;
+    closedir $dir_h;
     push @rm_dirs, $dir unless $is_needed;
     return $is_needed;
 }
